integration_tests/src/main/python/cast_test.py

```diff
@@ -167,3 +166,16 @@ def test_cast_long_to_decimal_overflow():
             f.col('a').cast(DecimalType(18, -1))),
         conf={'spark.sql.legacy.allowNegativeScaleOfDecimal': True})
 
+
+@pytest.mark.xfail(not is_before_spark_311(), reason="Only in Spark 3.1.1+ do we have incompatibility with negative decimal scale", raises=IllegalArgumentException)
+def test_cast_string_to_decimal_fail(spark_tmp_path):
+    if (is_before_spark_311()):
+        with_gpu_session(
+            lambda spark: unary_op_df(spark, StringGen("[0-9]{9}")).select(
+                f.col('a').cast(DecimalType(8, -3))), conf={'spark.sql.legacy.allowNegativeScaleOfDecimal': True})
+    else:
+        data_path = spark_tmp_path + '/PARQUET_DATA'
+        with_gpu_session(
+            lambda spark: unary_op_df(spark, StringGen("[0-9]{9}")).select(
```

Comment:
But this does not fail, so why is the test called `_fail`?  It just writes a file out to parquet??? why? Just skip the test if it is before spark 311.

Reply:
I have renamed it. It will throw an IllegalArgument exception which will cause the test to xfail. I don't want to skip it, I want to make sure it's falling back to the CPU. Is there a better way to do this?

Reply:
There are specific APIs for checking if something fell back to the CPU.  If that is what you want you should use that API.